processor/src/aux_table/mod.rs

```diff
@@ -0,0 +1,160 @@
+
+/// Auxiliary table for the VM's hasher, bitwise, and memory co-processor components.
+///
+/// This component is responsible for building a final execution trace from the stacked coprocessor
+/// execution traces.
+///
+/// The auxiliary trace table can be thought of as 4 stacked segments in the following form:
+/// * Hasher segment: contains the hasher trace and selector *
+/// This segment fills the first rows of the table up to the length of the hasher `trace_len`.
+/// - column 0: selector column with values set to ZERO
+/// - columns 1-17: execution trace of hasher coprocessor
+///
+/// * Bitwise segment: contains the bitwise trace and selectors *
+/// This segment begins at the end of the hasher segment and fills the next rows of the table for
+/// the `trace_len` of the bitwise coprocessor.
+/// - column 0: selector column with values set to ONE
+/// - column 1: selector column with values set to ZERO
+/// - columns 2-14: execution trace of bitwise coprocessor
+/// - columns 15-17: unused columns padded with ZERO
+///
+/// * Memory segment: contains the memory trace and selectors *
+/// This segment begins at the end of the bitwise segment and fills the next rows of the table for
+/// the `trace_len` of the memory coprocessor.
+/// - column 0-1: selector columns with values set to ONE
+/// - column 2: selector column with values set to ZERO
+/// - columns 3-16: execution trace of memory coprocessor
+/// - column 17: unused column padded with ZERO
+///
+/// * Final segment: unused *
+/// This segment begins at the end of the memory segment and fills the rest of the rows in the table
+/// up to the full length of the execution trace.
+/// - columns 0-2: selector columns with values set to ONE
+/// - columns 3-17: unused columns padded with ZERO
+///
+pub struct AuxTable {
+    trace: AuxTableTrace,
+}
+
+impl AuxTable {
+    // CONSTRUCTOR
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an [AuxTable] initialized with its co-processor components and an empty trace.
+    pub fn new(target_length: usize) -> Self {
+        // Allocate columns for the trace of the auxiliary table.
+        // note: it may be possible to optimize this by initializing with Felt::zeroed_vector,
+        // depending on how the compiler reduces Felt(0) and whether initializing here + iterating
+        // to update selector values is faster than using resize to initialize all values
+        let trace = (0..AUX_TRACE_WIDTH)
+            .map(|_| Vec::<Felt>::with_capacity(target_length))
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("failed to convert vector to array");
+
+        Self { trace }
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an execution trace of the top 16 stack slots and helper columns as a single array.
```

Comment:
Seems like a wrong comment got copied over here.

Reply:
Oops, not sure how that happened. Fixed now!